06_puppy/alfredxiao/store_test.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerSuite struct {
+	suite.Suite
+	store        Storer
+	storeFactory func() Storer
+}
+
+func (s *storerSuite) SetupTest() {
+	s.store = s.storeFactory()
+}
+
+func (s *storerSuite) TestCreatePuppyHappyCase() {
+	err := s.store.CreatePuppy(Puppy{ID: "1", Colour: "Black"})
+	s.NoError(err, "Happy case puppy creation")
+
+	p, _ := s.store.ReadPuppy("1")
```

Comment:
```go
p, err := s.store.ReadPuppy("1")
s.NoError(err)
//....
```
also for all below - don't swallow errors in tests especially when they are against different implementations.

Reply:
This is testing puppy creation, hence swallowing errors in puppy read. Otherwise, the test code can become more lengthy. Not sure which way is better.